client/pages/clubs/[id].tsx

```diff
@@ -1,22 +1,275 @@
-import React from "react";
-import { useRouter } from "next/router";
-import { useAppContext } from "../../context";
+import React from 'react'
+import styled from 'styled-components'
+import { useAppContext } from '../../context'
+import { colours, desktopFontSize, device, fontWeight, PageTitle } from '../../styles/'
+import { Club } from '../../context/'
+import { BasicEvent } from '../../context/Base'
+import moment from 'moment-timezone'
+import { useRouter } from 'next/router'
+
+const mobile = `425px`
+const tablet = `768px`
+
+interface BannerProps {
+  backgroundImageUrl: string
+}
+
+const largerThan = (size: string): string => `(min-width: ${size})`
+
+const Banner = styled.div<BannerProps>`
+  background-image: url(${({ backgroundImageUrl }) => backgroundImageUrl || ''});
+  background-position: center;
+  height: 20vh;
+
+  @media ${largerThan(tablet)} {
+    height: 30vh;
+  }
+`
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  margin: 0 auto 100px;
+
+  width: 85%;
+
+  @media ${largerThan(mobile)} {
+    width: 65%;
+  }
+`
 
 export default function ClubDetail() {
-  const { clubs } = useAppContext();
-  const router = useRouter();
+  const { clubs } = useAppContext()
+  const router = useRouter()
 
-  const { id } = router.query;
-  if (typeof id !== 'string') return null;
+  const { id } = router.query
+  if (typeof id !== 'string') return null
 
-  const { name, description } = clubs.get(parseInt(id));
+  const club = clubs.get(parseInt(id))
 
   return (
     <div>
-      <h1>Club Details</h1>
-      <p>{id}</p>
-      <p>{name}</p>
-      <p>{description}</p>
+      <Banner backgroundImageUrl={club.backgroundImageURL} />
+      <Container>
+        <ClubInfo club={club} />
+        <EventsHostedByClub events={club.events} />
+      </Container>
     </div>
-  );
+  )
+}
+
+const BackArrow = styled.img`
+  margin-top: 64px;
+  cursor: pointer;
+`
+
+const Logo = styled.img`
+  border-radius: 100%;
+  display: block;
+  margin-top: 40px;
+  height: 100px;
+  width: 100px;
+`
+
+const ClubName = styled(PageTitle)`
+  margin-bottom: 0;
+  margin-top: 32px;
+`
+
+const Links = styled.div`
+  display: flex;
+  margin-top: 16px;
+`
+
+const SocialLink = styled.a`
+  text-decoration: underline;
+`
+
+const ClubMetaData = styled.div`
+  align-items: center;
+  display: flex;
+  justify-content: space-between;
+  margin-top: 16px;
+`
+
+const MemberCount = styled.div`
+  font-style: italic;
+  font-weight: ${fontWeight.bold};
+`
+
+const CategoriesWrapper = styled.div`
+  display: flex;
+  margin-right: -4px;
+`
+
+const Category = styled.p`
+  border: 1px solid black;
+  border-radius: 15px;
+  margin: 0 4px;
+
+  padding: 4px 10px;
+  font-size: 14px;
```

Comment:
really nit, but we should probably use the font size constants from` typography.ts` so if we change font sizes in the future it'll be dynamically changed.

Reply:
great catch! i also changed another occurrence of hard-coded font size